### ISSUE TITLE
Updating alfen_wallbox sensor took longer than the scheduled update interval

### ISSUE THREAD

**Issue by pergolafabio** [leeyuentuen/alfen_wallbox#124]
Hi, today a new Alfen installed, first thing is offcourse testing thisn integration, i added succesfully, i can see everything

But i also see below in a loop, what sensor is giving issues here? I left a ping running, looks OK

```
2024-05-16 16:14:30.153 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-05-16 16:14:35.154 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-05-16 16:14:40.154 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-05-16 16:14:45.156 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-05-16 16:14:50.156 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-05-16 16:14:55.158 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-05-16 16:15:00.159 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
```

![image](https://github.com/leeyuentuen/alfen_wallbox/assets/43251136/415a36db-5d16-47d5-a478-53347f3beedb)

**Comment by pergolafabio** [comment]
Is there a sensor maybe that never can get updated "yet" since i didnt charge my car yet for first time?
The backoffice is not activated yet ...

**Comment by pergolafabio** [comment]
Indeed, can't change a single setting, tried like backlight or max current... It doesn't change...
I can change it with ace tool...

Maybe it's because the Alfen is not connected yet to the Backoffice and didn't use a charge yet...

But should be unrelated to change any setting?

**Comment by mathieudz** [comment]
I have this also, but not consistently. The integration works mostly, but at times very slow or and at times it just fails, so I add some repetitions in my automations. I thought of lowering the scan interval (configurable since recently), but I can't find how to do so on existing devices. Is this possible?
When trying to diagnose these errors, I noticed that while another connection was authenticated, all other connections got huge delays (or were just stuck). Maybe the integration is blocking itself sometimes?

**Comment by Nenodema** [comment]
> Indeed, can't change a single setting, tried like backlight or max current... It doesn't change... I can change it with ace tool...
> 
> Maybe it's because the Alfen is not connected yet to the Backoffice and didn't use a charge yet...
> 
> But should be unrelated to change any setting?

This seems to be related to issues #121 and #110. Downgrading to version 2.7.29 seems to be a good temporary workaround.

**Comment by pergolafabio** [comment]
ok, first charge completed, now the error is gone, probably it can now download the last charge transactions


But i"m still unable to change any values like max current

**Comment by pergolafabio** [comment]
ah wait , after changing a nubmer, it seems it takes a while before it updates...

**Comment by leeyuentuen** [comment]
> I have this also, but not consistently. The integration works mostly, but at times very slow or and at times it just fails, so I add some repetitions in my automations. I thought of lowering the scan interval (configurable since recently), but I can't find how to do so on existing devices. @leeyuentuen Is this possible? When trying to diagnose these errors, I noticed that while another connection was authenticated, all other connections got huge delays (or were just stuck). Maybe the integration is blocking itself sometimes?

problem is that alfen wallbox only accepts 1 connection simultaneously.

**Comment by leeyuentuen** [comment]
can you see if the latest 2.8.9 solved the problem?

**Comment by gigatexel** [comment]
I'm new here. I just installed the integration on an Alfen 7.0.0-4318

- Sensor values do not update (just a flatline) after initialisation of the integration
- I'm getting enormous amounts of "Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05" warnings

**Comment by leeyuentuen** [comment]
> I'm new here. I just installed the integration (latest public version) on an Alfen 7.0.0-4318
> 
> * Sensor values do not update (just a flatline) after initialisation of the integration
> * I'm getting enormous amounts of "Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05" warnings
which version are you running,

can you post more log from your HA?

**Comment by gigatexel** [comment]
Just endless stream

```
2024-08-09 15:39:15.454 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-08-09 15:39:20.455 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-08-09 15:39:25.456 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
2024-08-09 15:39:30.457 WARNING (MainThread) [homeassistant.components.sensor] Updating alfen_wallbox sensor took longer than the scheduled update interval 0:00:05
```

How do I increase the integration log level? I tried the following, but no success:
```
service: logger.set_level
data:
  custom_components.alfen_wallbox.alfen: debug
  custom_components.alfen_wallbox: debug
```

**Comment by Cornmeister** [comment]
I have the same issue, even after changing the update interval to 30 seconds or something, the logs keeps saying the interval is 5 seconds.

**Comment by Maes152** [comment]
I also have this issue. It doesn't even seem to update any of the sensor values.
I noticed that the integration in HACS says I've installed v2.8.9 but when I add the integration in HA it says 2.8.0. 
Completely removed the entire integration and reinstalled everything. Still the same issue. Log is just spammed with "Updating took longer than the scheduled update interval 0:00:05"

**Comment by Maes152** [comment]
Reverted Alfen charger to 6.6.2 and all is good again.

**Comment by grasmachien** [comment]
I had a lot of issues getting this integration to start.
Now it started I am getting this issue too...
Seems like this integration has a lot of bugs.
I have disabled the integration and am back to using the EVE Connect app.
It's a shame I can't track the charging in HA though.

The integration is able to connect but if I change the charging mode everything breaks down and I start getting the same `took longer than the scheduled update interval 0:00:05` message in the logs. I even set the interval to 30sec in the setup.
This thing is really unreliable.

**Comment by Mr-Star-Lord** [comment]
Unfortunately all versions after 2.7.29 don't work on my Home Assistant and Alfen Wallbox. 
Entities never update after initial bootup, unless I downgrade to 2.7.29.

Even with 2.8.9 still issues (which gave some issues to downgrade thanks to new HACS, but manually downgrade works).

My working setup:
HA 2024.10.4
Addon version: 2.7.29 
Wallbox: Alfen Eve Single Pro-line
Firmware: 6.4.0-4210

**Comment by fwestenberg** [comment]
Is this still an issue at the latest version? I found transaction data causes issues. In the latest version it can be deactivated from the options menu.
